rfcs/2020-10-29-3736-remap-chain-parsing.md

```diff
@@ -0,0 +1,181 @@
+
+
+## Alternatives
+
+### Enforce boolean conditions
+Allowing for non-boolean conditions in the `if` condition is not strictly necessary. A similar result could be achieved by running the regular
+expression twice, one with `match` which returns a boolean if it matches, and then again with `matches` which would extract the matches out of 
+the text.
+
+```coffeescript
+if match(.message, /^Started (?P<method>[^\s]*) for (?P<remote_addr>[^\s]*)/) {
+  $match = matches(.message, /^Started (?P<method>[^\s]*) for (?P<remote_addr>[^\s]*)/)
+  .method = $match.method
+  .remote_addr = $match.remote_addr
+  .source = "nginx"
+}
+```
+
+However, this has performance implications as the regular expression has to be run twice.
+
+### Use `switch`
+There are some alternative syntaxes that may be worth considering. One possibility that is common in many languages is a `switch` statement.
+
+```coffeescript
+switch {
+ ($match = matches(.message, /^Started (?P<method>[^\s]*) for (?P<remote_addr>[^\s]*)/)): {
+    .method = $match.method
+    .remote_addr = $match.remote_addr
+    .source = "nginx"
+  }
+ ($match = matches(.message, /^(?P<remote_addr>[^\s]*).*"(?P<method>[^\s]*).*"$/): {
+    .method = $match.method
+    .remote_addr = $match.remote_addr
+    .source = "haproxy"
+  }
+  default: {
+    .source = "other"
+  }
+}
+```
+
+The disadvantage of this approach is it is an extra syntactical element that needs to be added to the language that would require maintenance and
+make the language harder to learn. The advantage, however, is that we can then potentially avoid having to enable `if` statements to work with non-boolean 
+values. However, I'm not convinced this would be a good idea as having conditions working differently depending on their context could be confusing
+for the user.
+
+
+## Outstanding Questions
+
+- One thing that the `swimlanes` can do that this RFC hasn't catered for is that `swimlanes` can send the event to different sinks depending on a condition. 
+With Remap there is only one possible component, another transform or a sink, that can accept the output of the mapping. Is this something that should 
+be catered for by the Remap language and this RFC?
+
+
+## Plan Of Attack
+
+Incremental steps that execute this change. Generally this is in the form of:
+
+- [ ] Add `else if` to the remap language parser and add multiple branches to the `IfStatement` struct to handle multiple conditions.
+- [ ] Change the `IfStatement` functionality to treat non-boolean values as pass and fail for the conditional.
+- [ ] Enhance the remap parser to allow assignment within `if` statements. The underlying code already allows for assignments to 
+      pass on the assigned value to the outer expression, so no change should be necessary there.
+- [ ] Write the `matches` function that will run a regular expression and return any matched groups found. This is potentially out of scope
+      for this rfc, but would be useful nevertheless.
```

Comment:
I agree this is nice to have 👍 I also agree we can remove this from the RFC, and just go ahead and implement this.

There might at most be some debate on the function name (as it's close to the existing `match`, and Go and Rust use the term `find` or `find_captures`), but that's an easy change to make in an eventual final-pass before we release 0.12.